synthesizer/src/vm/finalize.rs

```diff
@@ -32,9 +32,11 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
     ///     `Ratify::BlockReward(block_reward)` and `Ratify::PuzzleReward(puzzle_reward)`
     ///     to the front of the `ratifications` list.
     #[inline]
+    #[allow(clippy::too_many_arguments)]
     pub fn speculate<'a, R: Rng + CryptoRng>(
         &self,
         state: FinalizeGlobalState,
+        time_since_last_block: i64, // TODO (raychu86): Consider moving this value into `FinalizeGlobalState`.
```

Comment:
Just a note: while it increases randomness of the seed, it also adds nontrivial computation because `hash_bhp768` is used in there, so I would then not add it to the preimage since we don't really need the pseudorandomness at this stage.

Reply:
It's possible to add to `FinalizeGlobalState` and not include it as part of the seed, but I refrained from doing something so confusing and inconsistent.